solc/CommandLineParser.cpp

```diff
@@ -941,9 +959,27 @@ void CommandLineParser::processArgs()
 	for (auto& option: conflictingWithStopAfter)
 		checkMutuallyExclusive({g_strStopAfter, option});
 
+	array<string, 11> const conflictingWithAsmJsonImport{
```

Comment:
@chriseth @ekpyron and what about this? does it make sense to not allow those?

Reply:
Again @cameel

Reply:
Not completely sure about `storageLayout`. So we lose that info at assembly level?
And `--gas` - so the estimator must be working at the Solidity level I assume and not by looking at opcodes?

By the way, putting it here seems redundant to me. You already do this check in `parseOutputSelection()` so this will never fail. `conflictingWithStopAfter` was here because it's just a regular option, not an input mode. And `conflictingWithStopAfter` should eventually be integrated into `validOptionInputModeCombinations` above anyway - this was being refactored in #12166 but it's stalled for now.